src/app/pages/Kafka/metricsApi.ts

```diff
@@ -0,0 +1,192 @@
+import { GetKafkaInstanceMetricsResponse, GetTopicsMetricsResponse } from '@rhoas/app-services-ui-components';
+import { TopicsApi } from '@rhoas/kafka-instance-sdk';
+import { Configuration, ConfigurationParameters, DefaultApi, RangeQuery } from '@rhoas/kafka-management-sdk';
+
+type NoUndefinedField<T> = {
+  [P in keyof T]-?: NoUndefinedField<NonNullable<T[P]>>;
+};
+type SafeRangeQuery = NoUndefinedField<RangeQuery>;
+
+export type TimeSeriesMetrics = { [timestamp: number]: number };
+export type PartitionBytesMetric = { [partition: string]: TimeSeriesMetrics };
+
+export type BasicApiConfigurationParameters = Pick<ConfigurationParameters, 'accessToken' | 'basePath'>;
+
+type FetchDiskSpaceMetricsProps = {
+  kafkaId: string;
+  duration: number;
+  interval: number;
+} & BasicApiConfigurationParameters;
+export const fetchKafkaInstanceMetrics = async ({
+  kafkaId,
+  duration,
+  interval,
+  accessToken,
+  basePath,
+}: FetchDiskSpaceMetricsProps): Promise<GetKafkaInstanceMetricsResponse> => {
+  const apisService = new DefaultApi(
+    new Configuration({
+      accessToken,
+      basePath,
+    })
+  );
+
+  const response = await apisService.getMetricsByRangeQuery(kafkaId, duration, interval, [
+    'kubelet_volume_stats_used_bytes',
+  ]);
+
+  // Remove all results with no data. Not sure this can really  happen but since
+  // the types allow for undefined we need to do a bit of defensive programming.
+  const safeMetrics: SafeRangeQuery[] = (response.data.items || []).filter(
+    (m) =>
+      // defensive programming
+      !(
+        m.values &&
+        m.metric &&
+        m.metric.topic &&
+        m.metric.name &&
+        m.metric.persistentvolumeclaim &&
+        m.metric.persistentvolumeclaim.includes('zookeeper')
+      )
+  ) as SafeRangeQuery[];
+
+  const usedDiskSpaceMetrics: TimeSeriesMetrics = {};
+
+  safeMetrics.forEach((m) => {
+    m.values.forEach(
+      ({ value, timestamp }) => (usedDiskSpaceMetrics[timestamp] = value + (usedDiskSpaceMetrics[timestamp] || 0))
+    );
+  });
+  return {
+    usedDiskSpaceMetrics,
+    clientConnectionsMetrics: [],
+    connectionAttemptRateMetrics: [],
```

Comment:
@ajaypratap003 this is where we need to fill the missing metrics for the 2 charts about the Kafka instance